### ISSUE TITLE
BUG: SpriteHandler error when dropping items B: 25

### ISSUE THREAD

**Issue by DooblyNoobly** [unitystation/unitystation#2221]
# Description

Sometimes when dropping items from your hand SpriteHandler throws an ArgumentOutOfRangeException.

Full error log:

```
ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
System.ThrowHelper.ThrowArgumentOutOfRangeException (System.ExceptionArgument argument, System.ExceptionResource resource) (at <599589bf4ce248909b8a14cbe4a2034e>:0)
System.ThrowHelper.ThrowArgumentOutOfRangeException () (at <599589bf4ce248909b8a14cbe4a2034e>:0)
System.Collections.Generic.List`1[T].get_Item (System.Int32 index) (at <599589bf4ce248909b8a14cbe4a2034e>:0)
SpriteHandler.PushTexture () (at Assets/Scripts/Sprite Handler/SpriteHandler.cs:63)
ClothingItem.PushTexture () (at Assets/Scripts/Items/ClothingItem.cs:169)
ClothingItem.SetReference (UnityEngine.GameObject Item) (at Assets/Scripts/Items/ClothingItem.cs:100)
EquipmentSpritesMessage+<Process>d__7.MoveNext () (at Assets/Scripts/Messages/Server/EquipmentSpritesMessage.cs:34)
```

## Steps to Reproduce
Please enter the steps to reproduce the bug or behaviour:
1. Start game from latest develop in the editor
2. Go to a locker with items inside, open it and pick up an item
3. Drop the item using shortcut key. Do a few times if it doesn't error right away it will happen pretty quickly

**Comment by DooblyNoobly** [comment]
nvm I can't recreate it. Must of been a meta data problem. Probably a very rare intermittent bug that won't show up in builds anyway. Closing

**Comment by fomalsd** [comment]
still happens for me occasionally

**Comment by fomalsd** [comment]
fresh example when picking up ID card from locker as client HoP
```
ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
  at System.ThrowHelper.ThrowArgumentOutOfRangeException (System.ExceptionArgument argument, System.ExceptionResource resource) [0x00029] in <599589bf4ce248909b8a14cbe4a2034e>:0 
  at System.ThrowHelper.ThrowArgumentOutOfRangeException () [0x00000] in <599589bf4ce248909b8a14cbe4a2034e>:0 
  at System.Collections.Generic.List`1[T].get_Item (System.Int32 index) [0x00009] in <599589bf4ce248909b8a14cbe4a2034e>:0 
  at SpriteHandler.PushTexture () [0x00070] in C:\homeproj\unitystation\UnityProject\Assets\Scripts\Sprite Handler\SpriteHandler.cs:63 
  at ClothingItem.PushTexture () [0x00012] in C:\homeproj\unitystation\UnityProject\Assets\Scripts\Items\ClothingItem.cs:173 
  at ClothingItem.SetReference (UnityEngine.GameObject Item) [0x00101] in C:\homeproj\unitystation\UnityProject\Assets\Scripts\Items\ClothingItem.cs:104 
  at PlayerAppearanceMessage+<Process>d__6.MoveNext () [0x000f4] in C:\homeproj\unitystation\UnityProject\Assets\Scripts\Messages\Server\PlayerAppearanceMessage.cs:48 
  at UnityEngine.SetupCoroutine.InvokeMoveNext (System.Collections.IEnumerator enumerator, System.IntPtr returnValueAddress) [0x00028] in C:\buildslave\unity\build\Runtime\Export\Scripting\Coroutines.cs:17 
```

**Comment by chairbender** [comment]
whoops, IDK why I marked that as fixed since I know it definitely wasn't, I think I typoed 2221 instead of 2211

**Comment by DooblyNoobly** [comment]
Contributor will receive a bounty of 25usd on a successful merge of a solution that resolves the issue above.

**Comment by JesterX666** [comment]
Little debugging tip:

Putting this code in an exception handler inside the PushTexture() function should break whenever it happens to facilitate debugging of what actually happens:

			System.Diagnostics.Debugger.Launch(); // Will attach the debugger if not already attached
			System.Diagnostics.Debugger.Break(); // Force break on this line

I'm putting it on my own branch, feel free to do the same and instruct us as "what was the cause"

Don't forget to remove those after finding it :)

**Comment by chairbender** [comment]
@JesterX666 for a unity specific one you can also even do "Debug.Break()" and it will actually pause the game so you can inspect the current state of the Scene in the editor

**Comment by chairbender** [comment]
@Bod9001 you mentioned this error no longer happens but underlying cause isn't fixed, what would you say is needed on this ticket still?

**Comment by Bod9001** [comment]
humm Could be just moved to a general cleanup of sprite handler?
I think another good suggestion would be moving over from sprite list to SpriteAndData And the corresponding Prefab changes needed for it?

**Comment by chairbender** [comment]
Then I propose we can just close this, consider https://github.com/unitystation/unitystation/pull/2482/commits/6ecb8f3b8d91b2ca64aa0c264627c6834bd46bb0 as having fixed it, and address this in #2311

**Comment by chairbender** [comment]
let's do that
